### PR TITLE
Fix #950: Restore elapsed tach hours in towplane logbook and mark future deadlines

### PR DESCRIPTION
**Pull request by pietbarber**
## Summary
Fixes Issue #950 by restoring elapsed tach-hour visibility in the towplane logbook when `tach_time` is missing, and improving deadline clarity by marking future deadlines as upcoming.

## What Changed
- Towplane closeouts now auto-compute `tach_time` from `end_tach - start_tach` when missing.
- Towplane logbook view now falls back to derived elapsed tach hours when `tach_time` is null.
- Deadline payloads now include an `is_future` flag.
- Equipment logbook template now visually mutes future deadlines and labels them as `Upcoming`.
- Added regression tests for:
  - elapsed tach fallback from start/end tach
  - future deadline upcoming flag behavior

## Validation
- `pytest logsheet/tests/test_towplane_logbook_performance.py -k "elapsed_tach_uses_start_end_when_tach_time_missing or future_deadline_is_marked_upcoming or deadline_appears_on_due_date" -q`
- `pytest logsheet/tests/test_towplane_logbook_performance.py logsheet/tests/test_manual_towplane_addition.py logsheet/tests/test_towplane_rental_setting.py -q`

## Issue
- Closes #950

### CODE REVIEW

File: logsheet/models.py [hunk 1]
```diff
@@ -1838,6 +1838,21 @@ def rental_cost_display(self):
...
+            self.tach_time is None
+            and self.start_tach is not None
+            and self.end_tach is not None
+        ):
+            start_tach = Decimal(str(self.start_tach))
+            end_tach = Decimal(str(self.end_tach))
+            self.tach_time = (end_tach - start_tach).quantize(
+                Decimal("0.01"),
+                rounding=ROUND_HALF_UP,
+            )
```
[the remark this answers is not in the document]
Comment (pietbarber) [reply logsheet/models.py:1855]: Implemented in logsheet/models.py: TowplaneCloseout.save() now clamps computed elapsed tach to non-negative values before quantizing. Specifically, when tach_time is auto-derived from end_tach - start_tach, we apply max(diff, 0.00) so bad readings (end_tach < start_tach) cannot persist negative elapsed time.

File: logsheet/views.py [hunk 2]
```diff
@@ -3808,11 +3814,19 @@ def towplane_logbook(request, pk: int):
     # Note: When multiple closeouts exist for the same day (e.g., from different
     # logsheets), we use the first logsheet's PK encountered. The closeouts are
     # ordered by log_date and logsheet_id, so this is deterministic.
+    def _elapsed_tach_hours(closeout):
+        if closeout.tach_time is not None:
+            return float(closeout.tach_time)
+        if closeout.start_tach is not None and closeout.end_tach is not None:
+            return float(closeout.end_tach - closeout.start_tach)
+        return 0.0
```
[the remark this answers is not in the document]
Comment (pietbarber) [reply logsheet/views.py:3822]: Implemented in logsheet/views.py: towplane_logbook() now clamps elapsed tach values in _elapsed_tach_hours() for both paths: (1) stored tach_time and (2) derived end_tach - start_tach fallback. This prevents negative elapsed tach from surfacing in legacy/invalid data scenarios and aligns behavior with tow_logbook utility clamping.

File: logsheet/tests/test_towplane_logbook_performance.py [hunk 3]
```diff
@@ -435,6 +435,39 @@ def test_multiple_closeouts_same_day(self):
...
+            created_by=self.member,
+        )
+
+        TowplaneCloseout.objects.create(
+            logsheet=logsheet,
+            towplane=self.towplane,
+            start_tach=100.0,
+            end_tach=101.7,
+            tach_time=None,
+        )
```
[the remark this answers is not in the document]
Comment (pietbarber) [reply logsheet/tests/test_towplane_logbook_performance.py]: Updated the test in logsheet/tests/test_towplane_logbook_performance.py so it truly exercises the view fallback path. The test now creates a closeout with start/end tach and then uses queryset update() to clear tach_time back to NULL (bypassing model save auto-computation), ensuring towplane_logbook() must derive elapsed hours from start/end values. Also added a regression test that verifies negative elapsed values are clamped to 0.0 in logbook output.

File: logsheet/models.py [hunk 4]
```diff
@@ -1838,6 +1838,22 @@ def rental_cost_display(self):
...
+        ):
+            start_tach = Decimal(str(self.start_tach))
+            end_tach = Decimal(str(self.end_tach))
+            elapsed_tach = max(end_tach - start_tach, Decimal("0.00"))
+            self.tach_time = elapsed_tach.quantize(
+                Decimal("0.01"),
+                rounding=ROUND_HALF_UP,
+            )
+        super().save(*args, **kwargs)
+
```
[the remark this answers is not in the document]
Comment (pietbarber) [reply logsheet/models.py:1864]: Implemented in logsheet/models.py: TowplaneCloseout.save() now handles save(update_fields=...) safely when tach_time is auto-derived. If tach_time is derived during save and update_fields is provided, the method now injects "tach_time" into update_fields before calling super().save(), ensuring the computed value is persisted instead of being dropped. Added regression coverage in logsheet/tests/test_towplane_logbook_performance.py (test_save_update_fields_persists_derived_tach_time) to verify this behavior.
